integration-tests/rest-tests-jaxrs/src/test/groovy/org/hawkular/metrics/rest/TagsITest.groovy

```diff
@@ -297,17 +297,27 @@ class TagsITest extends RESTTest {
 
       response = hawkularMetrics.get(path: it.path + "/tags/a1:*,d1:B%3AA",
           headers: [(tenantHeaderName): tenantId])
-      assertEquals(200, response.status)
 
-      assertEquals([
-                        a1: ['a', 'A/B'],
-                        d1: ['B:A']
-                    ], response.data)
+      def genericResponse = hawkularMetrics.get(path: "metrics/tags/a1:*,d1:B%3AA",
+          query: [type: it.type], headers: [(tenantHeaderName): tenantId])
+
+      def untypedResponse = hawkularMetrics.get(path: "metrics/tags/a1:*,d1:B%3AA", headers: [(tenantHeaderName):
+                                                                                                  tenantId])
+
+      [response, genericResponse, untypedResponse].each { lresponse ->
+        assertEquals(200, lresponse.status)
+        assertEquals([
+            a1: ['a', 'A/B'],
+            d1: ['B:A']
+        ], lresponse.data)
```

Comment:
No, the user does not get any metric name either, only the possible tag values. For this you could think that you want all the hostnames queried, but all of them don't have say gauge metric definitions. So just ask from all the metric types (as duplicates are removed).

In this test the reply is the same, but it would be different if there were different tag values on another type.